RELEASES_CHECKLIST.md

```diff
@@ -54,53 +54,58 @@ in the future.
 1. Open a release PR
     - Wait for approvals from Core team members.
     - Ensure the entire CI pipeline is green.
-1. Do a dry run with `cargo release [new_version] -v --no-tag --no-push`
-    - `[new_version]` should be the **exact** SemVer compatible version you are attempting 
-      to release e.g. `4.0.0-alpha.3`
-      - It is possible to supply a SemVer level here e.g. `major`, `minor`, `patch` or 
-        `<pre-release-name>`, however this will automatically bump and commit the changes 
-        to the `Cargo.toml` manifests. We have already done that in an earlier step so it 
+1. Before releasing, take note of the release order:
+    - `ink_e2e` is dependent on the `cargo-contract` crates, which are dependent on the `ink` "core" crates.
+    - Therefore, first release all `ink` crates except `ink_e2e`.
+    - If there are new versions of the `cargo-contract` crates, release them.
```

Comment:
In fact, there will necessarily be a release of `cargo-contract` crates which depend on the new version of `ink`. So those should be released first before `e2e`, then we can use the released version of `contract-build` there.

It will probably work without this for the RC release, but we will need to do this for the main release.